datajunction-server/datajunction_server/models/access.py

```diff
@@ -0,0 +1,436 @@
+"""
+Models for authorization
+"""
+from copy import deepcopy
+from enum import Enum
+from http import HTTPStatus
+from typing import (
+    TYPE_CHECKING,
+    Callable,
+    FrozenSet,
+    Iterable,
+    List,
+    Optional,
+    Set,
+    Union,
+)
+
+from pydantic import BaseModel, Field
+from sqlmodel import Session
+
+from datajunction_server.construction.utils import try_get_dj_node
+from datajunction_server.errors import DJError, DJException, ErrorCode
+from datajunction_server.models.node import Node, NodeRevision
+from datajunction_server.models.user import User
+
+if TYPE_CHECKING:
+    from datajunction_server.sql.parsing.ast import Column
+
+
+class ResourceRequestVerb(Enum):
+    """
+    Types of actions for a request
+    """
+
+    VIEW = "view"
+    READ = "read"
+    WRITE = "write"
+
+
+class ResourceObjectBase(BaseModel):
+    """
+    Base class for resource objects
+    that are passed to injected validation logic
+    """
+
+    name: str  # name of the node
+    created_by: str
```

Comment:
Does this need an `updated_by` as well? (Can be introduced later, of course)

Reply:
@shangyian this class exists as a layer between DJ internals and the injected code. The goal is to just provide some features the injected authorization code would want to look at to approve or deny access. You think they might want to know who updated for that?

Reply:
I see, in that case agreed that they shouldn't need updated_by. But then would it be clearer to just call this `owner` rather than `created_by`, since I assume that the owner of the node better represents access and may have changed from the user who created the node?